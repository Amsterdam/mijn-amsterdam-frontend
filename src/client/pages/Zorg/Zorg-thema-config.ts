import { WMOVoorzieningFrontend } from '../../../server/services/wmo/wmo-config-and-types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { MAX_TABLE_ROWS_ON_THEMA_PAGINA } from '../../config/app';
import styles from './Zorg.module.scss';

const MAX_TABLE_ROWS_ON_THEMA_PAGINA_HUIDIG = 5;
const MAX_TABLE_ROWS_ON_THEMA_PAGINA_EERDER = MAX_TABLE_ROWS_ON_THEMA_PAGINA;

const displayProps = {
  detailLinkComponent: '',
  status: 'Status',
  dateDescisionFormatted: 'Datum',
};

export const listPageParamKind = {
  actual: 'huidige-regelingen',
  historic: 'eerdere-en-afgewezen-regelingen',
} as const;

export type ListPageParamKey = keyof typeof listPageParamKind;
export type ListPageParamKind = (typeof listPageParamKind)[ListPageParamKey];

export const listPageTitle = {
  [listPageParamKind.actual]: 'Huidige regelingen',
  [listPageParamKind.historic]: 'Eerdere en afgewezen regelingen',
} as const;

export const tableConfig = {
  [listPageParamKind.actual]: {
    title: listPageTitle[listPageParamKind.actual],
    filter: (regeling: WMOVoorzieningFrontend) => regeling.isActual,
    sort: dateSort('dateDecision', 'desc'),
    displayProps: displayProps,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_HUIDIG,
    className: styles.HuidigeRegelingen,
  },
  [listPageParamKind.historic]: {
    title: listPageTitle[listPageParamKind.historic],
    filter: (regeling: WMOVoorzieningFrontend) => !regeling.isActual,
    sort: dateSort('dateDecision', 'desc'),
    displayProps: displayProps,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_EERDER,
    className: styles.EerdereRegelingen,
  },
} as const;

export const routes = {
  listPage: AppRoutes['ZORG/VOORZIENINGEN_LIST'],
  detailPage: AppRoutes['ZORG/VOORZIENING'],
  themaPage: AppRoutes['ZORG'],
} as const;
