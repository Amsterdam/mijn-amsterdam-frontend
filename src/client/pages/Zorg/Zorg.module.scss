.SectionCollapsibleCurrent {
  border-top: 0.1rem solid var(--ams-color-neutral-grey1);
  padding-top: 1.7rem;
}

@media screen and (min-width: $ams-breakpoint-medium) {
  .EerdereRegelingen,
  .HuidigeRegelingen {
    tr {
      > th,
      > td {
        &:nth-child(1) {
          width: 50%;
        }
        &:nth-child(2),
        &:nth-child(3) {
          width: 25%;
        }
      }
    }
  }
}
