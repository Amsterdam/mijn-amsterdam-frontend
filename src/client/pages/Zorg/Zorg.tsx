import { useMemo } from 'react';
import { AppRoutes, ChapterTitles } from '../../../universal/config';
import { isError, isLoading } from '../../../universal/helpers';
import {
  addTitleLinkComponent,
  ErrorAlert,
  ChapterIcon,
  Linkd,
  LinkdInline,
  OverviewPage,
  PageContent,
  PageHeading,
  SectionCollapsible,
  Table,
  MaintenanceNotifications,
} from '../../components';
import { ExternalUrls } from '../../config/app';
import { useAppStateGetter } from '../../hooks/useAppState';
import styles from './Zorg.module.scss';

const DISPLAY_PROPS = {
  title: '',
};

export default function Zorg() {
  const { WMO } = useAppStateGetter();

  const itemsActual = useMemo(() => {
    if (!WMO.content?.length) {
      return [];
    }
    return addTitleLinkComponent(WMO.content?.filter((item) => item.isActual));
  }, [WMO]);

  const itemsPrevious = useMemo(() => {
    if (!WMO.content?.length) {
      return [];
    }
    return addTitleLinkComponent(WMO.content?.filter((item) => !item.isActual));
  }, [WMO]);

  return (
    <OverviewPage className={styles.ZorgOverviewPage}>
      <PageHeading
        backLink={{
          to: AppRoutes.HOME,
          title: 'Home',
        }}
        icon={<ChapterIcon />}
      >
        {ChapterTitles.ZORG}
      </PageHeading>
      <PageContent>
        <p>
          Hieronder ziet u uw regelingen en hulpmiddelen vanuit de Wmo. Hebt u
          vragen of wilt u een wijziging doorgeven? Bel dan gratis de Wmo
          Helpdesk:{' '}
          <LinkdInline external={true} href="tel:08000643">
            0800 0643
          </LinkdInline>
          . Of ga langs bij het Sociaal Loket.
        </p>
        <p>
          <Linkd external={true} href={ExternalUrls.ZORG_LEES_MEER}>
            Lees hier meer over zorg en ondersteuning
          </Linkd>
        </p>
        <MaintenanceNotifications page="zorg" />
        {isError(WMO) && (
          <ErrorAlert>We kunnen op dit moment geen gegevens tonen.</ErrorAlert>
        )}
      </PageContent>
      <SectionCollapsible
        id="SectionCollapsible-healthcare-granted"
        title="Huidige regelingen en hulpmiddelen"
        noItemsMessage="U hebt geen huidige regelingen en hulpmiddelen."
        hasItems={!!itemsActual.length}
        startCollapsed={false}
        className={styles.SectionCollapsibleCurrent}
        isLoading={isLoading(WMO)}
      >
        <Table displayProps={DISPLAY_PROPS} items={itemsActual} />
      </SectionCollapsible>

      <SectionCollapsible
        id="SectionCollapsible-healthcare-previous"
        title="Eerdere regelingen en hulpmiddelen"
        noItemsMessage="U hebt geen eerdere regelingen en hulpmiddelen."
        hasItems={!!itemsPrevious.length}
        startCollapsed={true}
        isLoading={isLoading(WMO)}
      >
        <Table displayProps={DISPLAY_PROPS} items={itemsPrevious} />
      </SectionCollapsible>
      <p className={styles.HistoricItemsMention}>
        U ziet hier alleen informatie vanaf 1 januari 2018. Bel voor informatie
        van eerdere jaren de Wmo Helpdesk: 0800 0643.
      </p>
    </OverviewPage>
  );
}
