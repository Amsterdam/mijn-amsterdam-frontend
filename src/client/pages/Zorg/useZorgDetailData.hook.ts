import { useParams } from 'react-router-dom';

import { WMOVoorzieningFrontend } from '../../../server/services/wmo/wmo-config-and-types';
import { Themas } from '../../../universal/config/thema';
import { isLoading, isError } from '../../../universal/helpers/api';
import { useAppStateGetter } from '../../hooks/useAppState';
import { useThemaMenuItemByThemaID } from '../../hooks/useThemaMenuItems';

export function useZorgDetailData() {
  const { WMO } = useAppStateGetter();
  const { id } = useParams<{ id: WMOVoorzieningFrontend['id'] }>();
  const voorziening = WMO.content?.find((item) => item.id === id);
  const themaPaginaBreadcrumb = useThemaMenuItemByThemaID(Themas.ZORG);

  return {
    title: voorziening?.title ?? 'Voorziening',
    voorziening,
    themaPaginaBreadcrumb,
    isError: isError(WMO),
    isLoading: isLoading(WMO),
  };
}
