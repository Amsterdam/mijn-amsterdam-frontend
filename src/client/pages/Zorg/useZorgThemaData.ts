import {
  linkListItems,
  listPageParamKind,
  listPageTitle,
  routes,
  tableConfig,
} from './Zorg-thema-config';
import { WMOVoorzieningFrontend } from '../../../server/services/wmo/wmo-config-and-types';
import { isError, isLoading } from '../../../universal/helpers/api';
import { addLinkElementToProperty } from '../../components/Table/TableV2';
import { ThemaTitles } from '../../config/thema';
import { useAppStateGetter } from '../../hooks/useAppState';

export function useZorgThemaData() {
  const { WMO } = useAppStateGetter();

  const regelingen = addLinkElementToProperty<WMOVoorzieningFrontend>(
    WMO.content ?? [],
    'title',
    true
  );

  const title = ThemaTitles.ZORG;

  return {
    regelingen,
    title,
    isLoading: isLoading(WMO),
    isError: isError(WMO),
    routes,
    tableConfig,
    listPageTitle,
    listPageParamKind,
    linkListItems,
  };
}
