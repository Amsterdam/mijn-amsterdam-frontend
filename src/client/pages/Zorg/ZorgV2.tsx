import { Link, Paragraph } from '@amsterdam/design-system-react';
import { generatePath } from 'react-router-dom';

import { useZorgThemaData } from './useZorgThemaData';
import styles from './Zorg.module.scss';
import { WMOVoorzieningFrontend } from '../../../server/services/wmo/wmo-config-and-types';
import { LinkProps } from '../../../universal/types/App.types';
import { ExternalUrls } from '../../config/external-urls';
import ThemaPagina from '../ThemaPagina/ThemaPagina';
import ThemaPaginaTable from '../ThemaPagina/ThemaPaginaTable';

export default function ThemaPaginaZorg() {
  const { isError, isLoading, regelingen, title, routes, tableConfig } =
    useZorgThemaData();

  const pageContentTop = (
    <Paragraph>
      Hieronder ziet u uw voorzieningen vanuit de Wet maatschappelijke
      ondersteuning (Wmo). Heeft u vragen of wilt u een wijziging doorgeven? Bel
      dan gratis de Wmo Helpdesk:{' '}
      <Link rel="noreferrer" href="tel:08000643" variant="inline">
        0800 0643
      </Link>{' '}
      (maandag tot en met vrijdag van 08.00 tot 18.00 uur) of ga langs bij het
      Sociaal Loket.
    </Paragraph>
  );

  const linkListItems: LinkProps[] = [
    {
      to: ExternalUrls.ZORG_LEES_MEER,
      title: 'Lees hier meer over zorg en ondersteuning',
    },
  ];

  const tables = Object.entries(tableConfig).map(
    ([kind, { title, displayProps, filter, maxItems, className }]) => {
      return (
        <ThemaPaginaTable<WMOVoorzieningFrontend>
          key={kind}
          title={title}
          className={className}
          zaken={regelingen.filter(filter)}
          listPageRoute={generatePath(routes.listPage, {
            kind,
          })}
          displayProps={displayProps}
          textNoContent={`U heeft geen ${title.toLowerCase()}`}
          maxItems={maxItems}
        />
      );
    }
  );

  return (
    <>
      <ThemaPagina
        title={title}
        pageContentTop={pageContentTop}
        linkListItems={linkListItems}
        pageContentMain={tables}
        isError={isError}
        isLoading={isLoading}
      />
      <p className={styles.HistoricItemsMention}>
        U ziet hier informatie vanaf 1 januari 2018. Wilt u iets weten van
        daarvoor? Bel dan de Wmo Helpdesk:{' '}
        <Link href="tel:08000643">0800 0643</Link>.
      </p>
    </>
  );
}
