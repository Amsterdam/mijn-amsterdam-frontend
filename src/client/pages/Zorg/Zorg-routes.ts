import { ZorgThemaPagina } from './Zorg';
import { ZorgDetail } from './ZorgDetail';
import { ZorgRegelingen } from './ZorgRegelingen';
import { FeatureToggle } from '../../../universal/config/feature-toggles';

export const ZorgRoutes = [
  { route: '/zorg-en-ondersteuning/voorziening/:id', Component: ZorgDetail },
  {
    route: '/zorg-en-ondersteuning/lijst/:kind/:page?',
    Component: ZorgRegelingen,
    isActive: FeatureToggle.zorgv2ThemapaginaActive,
  },
  {
    route: '/zorg-en-ondersteuning',
    Component: ZorgThemaPagina,
    isActive: FeatureToggle.zorgv2ThemapaginaActive,
  },
];
