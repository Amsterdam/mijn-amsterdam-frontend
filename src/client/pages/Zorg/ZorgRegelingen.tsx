import { useParams } from 'react-router-dom';

import { useZorgThemaData } from './useZorgThemaData';
import { HistoricItemsMention } from './Zorg';
import { ListPageParamKind } from './Zorg-thema-config';
import { ListPagePaginated } from '../../components/ListPagePaginated/ListPagePaginated';
import { PageContentCell } from '../../components/Page/Page';
import { NotFound } from '../NotFound/NotFound';

export function ZorgRegelingen() {
  const { kind } = useParams<{ kind: ListPageParamKind }>();
  const {
    regelingen,
    tableConfig,
    routes,
    isLoading,
    isError,
    listPageParamKind,
  } = useZorgThemaData();
  const listPageTableConfig = tableConfig[kind];

  if (!listPageTableConfig) {
    return <NotFound />;
  }

  return (
    <ListPagePaginated
      items={regelingen.filter(listPageTableConfig.filter)}
      title={listPageTableConfig.title}
      appRoute={routes.listPage}
      appRouteParams={{ kind }}
      appRouteBack={routes.themaPage}
      displayProps={listPageTableConfig.displayProps}
      isLoading={isLoading}
      isError={isError}
      tableClassName={listPageTableConfig.className}
      pageContentBottom={
        <PageContentCell spanWide={8} startWide={3}>
          {kind === listPageParamKind.historic && <HistoricItemsMention />}
        </PageContentCell>
      }
    />
  );
}
