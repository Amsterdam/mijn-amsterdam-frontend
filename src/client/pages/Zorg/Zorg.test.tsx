import { render } from '@testing-library/react';

import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';
import { AppRoutes } from '../../../universal/config/routes';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import Zorg from './Zorg';

const testState: any = {
  WMO: {
    status: 'OK',
    content: [
      {
        id: 'wmo-item-1',
        title: 'Wmo item 1',
        supplier: 'Mantelzorg B.V',
        isActual: true,
        link: {
          to: 'http://example.org/ding',
          title: 'Linkje!! naar wmo item 1',
        },
        steps: [
          {
            id: 'wmo-step-1',
            status: 'Levering gestart',
            datePublished: '2020-07-24',
            description: 'De levering van uw thuizorg is gestart',
            documents: [],
            isActive: true,
            isChecked: true,
          },
        ],
      },
    ],
  },
};

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

describe('<Zorg />', () => {
  const routeEntry = generatePath(AppRoutes.ZORG);
  const routePath = AppRoutes.ZORG;

  const Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={Zorg}
      initializeState={initializeState}
    />
  );

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
