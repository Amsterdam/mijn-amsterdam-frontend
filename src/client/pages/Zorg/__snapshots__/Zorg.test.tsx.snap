// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Zorg /> > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <main
    class="_Page_47d7c0 _OverviewPage_47d7c0 _ZorgOverviewPage_14a7ab"
  >
    <header
      class="_PageHeading_3b337d _hasIcon_3b337d _hasBackLink_3b337d"
    >
      <span
        aria-hidden="true"
        class="_Icon_3b337d"
      >
        <svg
          aria-label="Zorg en ondersteuning"
          class="_ThemaIcon_2fc54d"
          fill="black"
          focusable="false"
          height="32px"
          viewBox="0 0 32 32"
          width="32px"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <title>
            zorg-en-ondersteuning-icoon
          </title>
          <polygon
            id="Path"
            points="14 25 18 25 18 21 22 21 22 17 18 17 18 13 14 13 14 17 10 17 10 21 14 21"
          />
          <path
            d="M25,6 L25,0 L7,0 L7,6 L0,6 L0,32 L32,32 L32,6 L25,6 Z M11,4 L21,4 L21,6 L11,6 L11,4 Z M28,28 L4,28 L4,10 L28,10 L28,28 Z"
            id="Shape"
          />
        </svg>
      </span>
      <div
        class="_HeadingInner_3b337d"
      >
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45 _BackLink_3b337d"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-links-icoon
            </title>
            <polygon
              points="22.8571429 32 6.85714286 16 22.8571429 0 25.7568858 2.91002571 12.6771943 16 25.7568858 29.0899743"
            />
          </svg>
          Home
        </a>
        <h2
          class="ams-heading ams-heading--1"
        >
          Zorg en ondersteuning
        </h2>
      </div>
    </header>
    <div
      class="_PageContent_47d7c0"
    >
      <p>
        Hieronder ziet u uw regelingen en hulpmiddelen vanuit de Wmo. Hebt u vragen of wilt u een wijziging doorgeven? Bel dan gratis de Wmo Helpdesk: 
        <a
          class="_Button_c3ab45 _Button__inline_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45 _LinkedInline_c3ab45"
          href="tel:08000643"
          rel="external noopener noreferrer"
        >
          0800 0643
        </a>
        . Of ga langs bij het Sociaal Loket.
      </p>
      <p>
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45"
          href="https://www.amsterdam.nl/zorg-ondersteuning/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
          Lees hier meer over zorg en ondersteuning
        </a>
      </p>
    </div>
    <section
      class="_SectionCollapsible_b93dc7 _SectionCollapsibleCurrent_14a7ab"
    >
      <h3
        class="ams-heading ams-heading--3 _Title_b93dc7 _TitleWithItems_b93dc7"
      >
        <button
          aria-expanded="true"
          class="_TitleToggle_b93dc7"
        >
          <svg
            aria-hidden="true"
            class="_CaretIcon_b93dc7"
            focusable="false"
            height="32px"
            viewBox="0 0 32 32"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
           Huidige regelingen en hulpmiddelen
        </button>
      </h3>
      <div
        aria-hidden="false"
        class="_Panel_b93dc7"
        style="height: 0px;"
      >
        <div
          class="_PanelInner_b93dc7"
        >
          <table
            class="_Table_a66dcc"
          >
            <tbody>
              <tr
                class="_TableRow_a66dcc"
              >
                <td
                  class="_DisplayPropTitle_a66dcc"
                >
                  <a
                    class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45"
                    href="/http://example.org/ding"
                  >
                    <svg
                      aria-hidden="true"
                      class="_Icon_c3ab45 _Icon__left_c3ab45"
                      fill="#000"
                      focusable="false"
                      height="14px"
                      viewBox="0 0 32 32"
                      width="14px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <title>
                        chevron-rechts-icoon
                      </title>
                      <polygon
                        points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
                      />
                    </svg>
                    Wmo item 1
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </section>
    <section
      class="_SectionCollapsible_b93dc7"
    >
      <h3
        class="ams-heading ams-heading--3 _Title_b93dc7 _TitleWithItems_b93dc7"
      >
        <button
          aria-expanded="false"
          class="_TitleToggle_b93dc7"
        >
          <svg
            aria-hidden="true"
            class="_CaretIcon_b93dc7"
            focusable="false"
            height="32px"
            viewBox="0 0 32 32"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
           Eerdere regelingen en hulpmiddelen
        </button>
      </h3>
      <div
        aria-hidden="true"
        class="_Panel_b93dc7"
        style="height: 0px;"
      >
        <div
          class="_PanelInner_b93dc7"
        >
          <p
            class="_NoItemsMessage_b93dc7"
          >
            U hebt geen eerdere regelingen en hulpmiddelen.
          </p>
        </div>
      </div>
    </section>
    <p
      class="_HistoricItemsMention_14a7ab"
    >
      U ziet hier alleen informatie vanaf 1 januari 2018. Bel voor informatie van eerdere jaren de Wmo Helpdesk: 0800 0643.
    </p>
  </main>
</DocumentFragment>
`;
